### PR TITLE
Issue 28 feat/api user service

### PR DESCRIPTION
**Pull request by Mariusz-web**

### CODE REVIEW

File: src/user/user.controller.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    async updateUser(req: Request, res: Response) {
+        const body = plainToInstance(UpdateUserDto, req.body as UpdateUserDto);
+        const errors = await validate(body);
+        if (errors.length > 0) {
+            return res.status(400).json(errors);
+        }
+
+        const userToUpdate = body;
```
Comment (Veryyapeee) [inline src/user/user.controller.ts:40]: what for is this constant? just put the body in `UserService.updateUser`

File: src/user/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+        return user;
+    }
+
+    async updateUser(userId: string, userData: UpdateUserDto) {
+        const userRepo = getRepository(User);
+
+        const user = await userRepo.findOne(userId);
+
+        if (!user) return null;
```
Comment (Veryyapeee) [inline src/user/user.service.ts:21]: use `if(){}` syntax

File: src/user/user.service.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { UpdateUserDto } from './dto/updateUser.dto';
+import { User } from './user.entity';
+
+export const UserService = new (class {
+    async getUser(userId: string) {
+        const userRepo = getRepository(User);
+
+        const user = await userRepo.findOne(userId);
+
+        if (!user) return null;
```
Comment (Veryyapeee) [inline src/user/user.service.ts:11]: same here about the syntax

File: yarn.lock [hunk 4]
```diff
@@ -4,26 +4,26 @@
```
Comment (PoProstuMieciek) [inline yarn.lock:4]: Don't use NPM! Please, revert changes to this file.

**Review by PoProstuMieciek — CHANGES_REQUESTED** [review]

**Review by PoProstuMieciek — APPROVED** [review]

**Review by Veryyapeee — APPROVED** [review]

**Pull request merged**